pkg/test/externalbinary/provider.go

```diff
@@ -0,0 +1,194 @@
+		}
+		cleanOldCacheFiles(cacheBase, logger)
+		binDir = path.Join(cacheBase, binDir)
+		logger.Printf("External binary cache is enabled, will use %q", cacheBase)
+	} else {
+		logger.Printf("External binary cache is disabled, using a temp directory instead")
+		var err error
+		tmpDir, err = os.MkdirTemp("", "openshift-tests")
+		if err != nil {
+			return nil, errors.Wrap(err, "couldn't create temp directory")
+		}
+		binDir = path.Join(tmpDir, binDir)
+	}
+	logger.Printf("Using path for binaries %q", binDir)
+
+	if err := createBinPath(binDir); err != nil {
+		return nil, errors.WithMessagef(err, "error creating cache path %q", binDir)
+	}
+
+	releasePayloadImageStream, releaseImage, err := extractReleaseImageStream(logger, binDir,
+		releaseImage, registryAuthfilePath)
+	if err != nil {
+		return nil, errors.WithMessage(err, "couldn't extract release payload image stream")
+	}
+
+	return &ExternalBinaryProvider{
+		registryAuthFilePath: registryAuthfilePath,
+		logger:               logger,
+		oc:                   oc,
+		imageStream:          releasePayloadImageStream,
+		binPath:              binDir,
+		tmpDir:               tmpDir,
+	}, nil
+}
+
+func (provider *ExternalBinaryProvider) Cleanup() {
+	if provider.tmpDir != "" {
+		if err := os.RemoveAll(provider.tmpDir); err != nil {
+			provider.logger.Printf("Failed to remove tmpDir %q: %v", provider.tmpDir, err)
+		} else {
+			provider.logger.Printf("Successfully removed tmpDir %q", provider.tmpDir)
+		}
+	}
+
+	provider.tmpDir = ""
+	provider.binPath = ""
+}
+
+// ExtractBinaryFromReleaseImage resolves the tag from the release image and extracts the binary,
+// checking if the binary is compatible with the current systems' architecture. It returns an error
+// if extraction fails or if the binary is incompatible.
+//
+// Note: When developing openshift-tests on a non-Linux non-AMD64 computer (i.e. on Apple Silicon), external
+// binaries won't work.  You would need to run it in a Linux environment (VM or container), and even then
+// override the payload selection with an aarch64 payload unless x86 emulation is enabled.
+func (provider *ExternalBinaryProvider) ExtractBinaryFromReleaseImage(tag, binary string) (*TestBinary, error) {
+	if provider.binPath == "" {
+		return nil, fmt.Errorf("extraction path is not set, cleanup was already run")
+	}
+
+	// Resolve the image tag from the image stream.
+	image := ""
+	for _, t := range provider.imageStream.Spec.Tags {
+		if t.Name == tag {
+			image = t.From.Name
+			break
+		}
+	}
+
+	if len(image) == 0 {
+		return nil, fmt.Errorf("%s not found", tag)
+	}
+
+	// Define the path for the binary
+	binPath := filepath.Join(provider.binPath, strings.TrimSuffix(filepath.Base(binary), ".gz"))
+
+	// Check if the binary already exists in the path
+	if _, err := os.Stat(binPath); err == nil {
+		provider.logger.Printf("Using existing binary %q for tag %q", binPath, tag)
+		return &TestBinary{
```

Comment:
You could keep the cached version fresh each time it is utilized, as a refinement. 
```
err := os.Chtimes(binPath, currentTime, currentTime)
```

Reply:
It does keep them fresh -- it's a side effect of running MkdirAll on the payload dir in cache path, and the cache cleaning only looking at top level entries (we keep or delete payload dirs not the binaries in them).

I'll probably add this anyway: belt and suspenders, and make the intention more obvious